CMakeLists.txt

```diff
@@ -185,6 +186,10 @@ add_subdirectory(share)
 add_subdirectory(src/framework/global) # should be first to work pch
 add_subdirectory(src)
 
+if (MUE_BUILD_LIVEBRAILLE_MODULE)
+    add_subdirectory(thirdparty/liblouis)
+endif(MUE_BUILD_LIVEBRAILLE_MODULE)
```

Comment:
It's better to add this 3rd party library in the same way as others are added - they are added by the module that uses it.
This has advantages:
* We can see exactly who is using a third-party library
* When we disable the target module, its dependencies are also automatically disabled

Reply:
That makes sense, but the module is not going to be part of this PR. @RomanPudashkin requested that the library go in first to make PR #16699 easier to review.

I could create a CMakeLists.txt at the location where the module will go, but I suspect that PR #16699 has put the files in the wrong places.

Where is the right place for the module?

- src/livebraille
- src/framework/livebraille
- src/notation/livebraille
- Somewhere else?

FYI, here's what the module will eventually do:

1. If the user selects an element in the score, the element's parent measure will be displayed as Braille in the viewer panel (this is implemented PR [#16699](https://github.com/musescore/MuseScore/pull/16699)).
2. Within the Braille string, the characters corresponding to the selected score element will be highlighted.
3. If the user clicks on a different Braille character, the corresponding element will become selected in the score.
4. If the user types some Braille into the viewer using some special keyboard shortcuts to represent Braille dots, this will be turned into ordinary notation in the score.

Also, the new module's code will be used to export Braille files instead of the current Braille export code. (I guess there would still be an export module but internally it would just call methods from the new module do the conversion.)

Reply:
@igorkorsukov, if you're not sure where the module should go then it might be best to merge this as it is so that PR [#16699 ](https://github.com/musescore/MuseScore/pull/16699) can be rebased and reviewed. I can always move these lines into the module as part of that PR once somebody has reviewed it and determined the proper place to put the module.

Reply:
@shoogle thanks for the clarifications  

I think this new module should be located in `src/livebraille`, like `palette`, `inspector`...
I would prefer this structure: 
```
src/livebraille/
                        internal/
                                      ...
                        thirdparty/ 
                                          .untidy
                                          liblouis/
                                                        ...
                        livebraillemodule.cpp
                        livebraillemodule.h
                        ...
```

I would prefer third-party libraries to also located into modules that require them. Previese to this, this was not done due to the fact that it was difficult to exclude them from code style checks. Now it's easy to do, and I plan to move them...

Reply:
I decided to name the module `braille` rather than `livebraille` because the code will also be used by the Braille Export module ([`iex_braille`](https://github.com/musescore/MuseScore/blob/4d8711c157170ad1d47696f9a39c6cd4d4401d33/src/importexport/braille/CMakeLists.txt#L21)).